nav2_rviz_plugins/src/navigation_dialog.cpp

```diff
@@ -130,19 +130,22 @@ NavigationDialog::startNavigation(double x, double y, double theta, std::string
   send_goal_options.result_callback = [](auto) {};
 
   auto future_goal_handle = action_client_->async_send_goal(goal_, send_goal_options);
-  if (rclcpp::spin_until_future_complete(client_node_, future_goal_handle) !=
+  if (rclcpp::spin_until_future_complete(client_node_,
```

Comment:
What's special about 5? This should be parameterized

Reply:
Nothing is special about 5. I don't think there is any perfect number for these timeouts. I will make it a parameter. Should there be two separate parameters for each timeout?

Reply:
It looks like both of thost `5`s are accomplishing the same thing: checking if we are able to get some basic response from the action server, so I think 1 parameter is all that is needed.

Make sure when you add a parameter you add the field to the nav2_params file so that we have a fully-described parameter file, there's a number of them missing right now (and on my queue to deal with) but it would make my life easier if it more were there than not.

Reply:
Thank you for your help in my on-going war against magic numbers

![image](https://user-images.githubusercontent.com/14944147/61393196-4db72900-a875-11e9-9a1a-be13901f35b6.png)

Reply:
haha.. Thanks for your guidance. From now on, I will be after the magic numbers in our stack :)

Reply:
@SteveMacenski Regarding your full-described parameter file...  FYI - I've written a parameter dumping utility. One can launch the nav2 stack and then invoke startup to take all of the nodes through configure and activate states. Then, run this utility. It will iterate over all of the nodes and dump every parameter in either MD or YAML formats, as specified with a command-line parameter. It's useful to document the default parameter values. I was thinking that one could start the stack without any nav2_params.yaml file and then dump the parameters. We'd probably have to specify the BT navigation XML file and map file from the launch script in this case.

Reply:
@SteveMacenski I left the timeouts as hard coded values for now. I think we agreed on we should parameterize all the hard codded magic numbers in the stack during our previous weekly work group meeting, but we haven't decided how we are going to do these changes yet. Since I think the Rviz crashing issue can be a blocking problem for some users, we can merge this PR if everything looks good in my changes. I will submit a different issue "parameterize all the hard coded values in the stack" for this discussion.